staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -295,45 +415,17 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 // in a retry-if-conflict loop.
 type nodeModifier func(*v1.Node)
 
-func (cnc *CloudNodeController) UpdateCloudNode(ctx context.Context, _, newObj interface{}) {
-	node, ok := newObj.(*v1.Node)
-	if !ok {
-		utilruntime.HandleError(fmt.Errorf("unexpected object type: %v", newObj))
-		return
-	}
-
-	cloudTaint := getCloudTaint(node.Spec.Taints)
-	if cloudTaint == nil {
-		// The node has already been initialized so nothing to do.
-		return
-	}
-
-	cnc.initializeNode(ctx, node)
-}
-
-// AddCloudNode handles initializing new nodes registered with the cloud taint.
-func (cnc *CloudNodeController) AddCloudNode(ctx context.Context, obj interface{}) {
-	node := obj.(*v1.Node)
-
-	cloudTaint := getCloudTaint(node.Spec.Taints)
-	if cloudTaint == nil {
-		klog.V(2).Infof("This node %s is registered without the cloud taint. Will not process.", node.Name)
-		return
-	}
-
-	cnc.initializeNode(ctx, node)
-}
-
-// This processes nodes that were added into the cluster, and cloud initialize them if appropriate
-func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Node) {
-	klog.Infof("Initializing node %s with cloud provider", node.Name)
+// syncNode handles updating existing nodes registered with the cloud taint
+// and processes nodes that were added into the cluster, and cloud initialize them if appropriate.
+func (cnc *CloudNodeController) syncNode(ctx context.Context, nodeName string) error {
+	klog.Infof("Syncing node %s", nodeName)
 
 	err := clientretry.RetryOnConflict(UpdateNodeSpecBackoff, func() error {
```

Comment:
This call to update node conditions for GCE should be moved after the cloud taint check

Reply:
@andrewsykim If I just move the check like this, the unit test `TestGCECondition` is always failing, the node's status won't update correspondingly, am I missing something? 🤔 
```
klog.Infof("Syncing node %s", nodeName)

	curNode, err := cnc.nodeInformer.Lister().Get(nodeName)
	if err != nil {
		if apierrors.IsNotFound(err) {
			return nil
		}

		return err
	}

	cloudTaint := getCloudTaint(curNode.Spec.Taints)
	if cloudTaint == nil {
		// Node object received from event had the cloud taint but was outdated,
		// the node has actually already been initialized, so this sync event can be ignored.
		return nil
	}

	err = clientretry.RetryOnConflict(UpdateNodeSpecBackoff, func() error {
...
}
```

Reply:
I think the nodeLister is not being set in the controller on line 1369 of `node_controller_test.go `

Reply:
@andrewsykim I think you are right, I wasn't fully&correctly set up the lister in `Test_syncNode()` so when I copy and paste it into these two unit tests, it's always failing